packit_service/worker/helpers/testing_farm.py

```diff
@@ -60,40 +61,72 @@ class CommentArguments:
     Parse arguments from trigger comment and provide the attributes to Testing Farm helper.
     """
 
-    packit_command: str = None
-    identifier: str = None
-    labels: List[str] = None
-    pr_argument: str = None
-
     def __init__(self, command_prefix: str, comment: str):
+        self.packit_command: str = None
+        self.identifier: str = None
+        self.labels: List[str] = None
+        self.pr_argument: str = None
+        self.envs: Dict[str, str] = None
+
         if comment is None:
             return
 
-        # Try to parse identifier argument from comment
+        # Remove the command prefix from the comment
         logger.debug(f"Parsing comment -> {comment}")
         logger.debug(f"Used command prefix -> {command_prefix}")
 
-        match = re.search(
-            r"^" + re.escape(command_prefix) + r"\s(?P<packit_command>\S+)", comment
-        )
+        # Match the command prefix and extract the rest of the comment
+        match = re.match(r"^" + re.escape(command_prefix) + r"\s+(.*)", comment)
         if match:
-            self.packit_command = match.group("packit_command")
-            logger.debug(f"Parsed packit_command: {self.packit_command}")
+            arguments_str = match.group(1)
+        else:
+            # If the command prefix is not found, nothing to parse
+            logger.debug("Command prefix not found in the comment.")
+            return
 
-        match = re.search(r"(--identifier|--id|-i)[\s=](?P<identifier>\S+)", comment)
-        if match:
-            self.identifier = match.group("identifier")
-            logger.debug(f"Parsed test argument -> identifier: {self.identifier}")
+        # Use shlex to split the arguments string into a list
+        args_list = shlex.split(arguments_str)
+        logger.debug(f"Arguments list after shlex splitting: {args_list}")
 
-        match = re.search(r"--labels[\s=](?P<labels>\S+)", comment)
-        if match:
-            self.labels = match.group("labels").split(",")
-            logger.debug(f"Parsed test argument -> labels: {self.labels}")
+        # Set up argparse
+        parser = argparse.ArgumentParser()
+        parser.add_argument("packit_command")
+        parser.add_argument("--identifier", "--id", "-i")
+        parser.add_argument("--labels")
+        parser.add_argument("--env", action="append")  # Allows multiple --env arguments
```

Comment:
maybe we could create property out of this, or at least separate the creation of the parser

Reply:
I am not python expert so maybe my changes are not python compliant, but I think it is better now